From eba21b9337443e01252ee34b203dc674ec70eae9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Boisnard?= <fredericx.boisnard@intel.com>
Date: Thu, 6 Sep 2012 15:16:13 +0200
Subject: [PORT FROM MAIN] PFW: Fix compilation issues on Ubuntu
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

BZ: 44356

The header file "intel_sst_ioctl.h" has recently been copied from the kernel
folder to the bionic folder, but the definition of "__packed" is still missing for
userland programs (namely, the parameter-framework).
This patch replaces all the occurences of "__packed" by "__attribute__ ((packed))"
in this file.

Change-Id: I560703819042a0f35acf0208b2d6bc5a455a7ee8
Orig-Change-Id: If5ab14fedc8223790fd49181699e4788e5517ef4
Signed-off-by: Frédéric Boisnard <fredericx.boisnard@intel.com>
Reviewed-on: http://android.intel.com:8080/75550
Reviewed-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Tested-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 libc/kernel/common/linux/sound/intel_sst_ioctl.h |   18 +++++++++---------
 1 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/libc/kernel/common/linux/sound/intel_sst_ioctl.h b/libc/kernel/common/linux/sound/intel_sst_ioctl.h
index 6822a35..97032db 100644
--- a/libc/kernel/common/linux/sound/intel_sst_ioctl.h
+++ b/libc/kernel/common/linux/sound/intel_sst_ioctl.h
@@ -97,7 +97,7 @@ struct snd_pcm_params_mrfld {
  __u16 reserved3;
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
  __u8 channel_map[8];
-} __packed;
+} __attribute__ ((packed));
 struct snd_pcm_params {
  __u16 codec;
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
@@ -189,7 +189,7 @@ struct snd_ppp_params {
  __u8 reserved;
  __u32 size;
  void *params;
-} __packed;
+} __attribute__ ((packed));
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
 struct snd_sst_postproc_info {
  __u32 src_min;
@@ -214,7 +214,7 @@ struct snd_sst_prp_info {
  __u8 volume_control;
  __u8 reserved1;
  __u16 reserved2;
-} __packed;
+} __attribute__ ((packed));
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
 struct snd_sst_ppp_info {
  __u32 src:1;
@@ -261,7 +261,7 @@ union snd_sst_codec_params {
 struct snd_sst_stream_params {
  union snd_sst_codec_params uc;
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
-} __packed;
+} __attribute__ ((packed));
 struct snd_sst_params {
  __u32 result;
  __u32 stream_id;
@@ -291,7 +291,7 @@ struct snd_sst_pmic_config {
  __u16 num_chan;
  __u16 pcm_wd_sz;
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
-} __packed;
+} __attribute__ ((packed));
 struct snd_sst_get_stream_params {
  struct snd_sst_params codec_params;
  struct snd_sst_pmic_config pcm_params;
@@ -352,13 +352,13 @@ struct snd_sst_slot_info {
  __u8 reserved;
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
  struct snd_sst_pmic_config pcm_params;
-} __packed;
+} __attribute__ ((packed));
 #define SST_MAX_TARGET_DEVICES 3
 struct snd_sst_target_device {
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
  __u32 device_route;
  struct snd_sst_slot_info devices[SST_MAX_TARGET_DEVICES];
-} __packed;
+} __attribute__ ((packed));
 struct snd_sst_driver_info {
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
  __u32 version;
@@ -413,7 +413,7 @@ struct snd_sst_tuning_params {
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
  __u8 rsvd;
  __u64 addr;
-} __packed;
+} __attribute__ ((packed));
 struct snd_sst_runtime_params {
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
  __u8 type;
@@ -422,7 +422,7 @@ struct snd_sst_runtime_params {
  __u8 rsvd;
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
  void *addr;
-} __packed;
+} __attribute__ ((packed));
 #define SNDRV_SST_STREAM_SET_PARAMS _IOWR('L', 0x00,   struct snd_sst_stream_params *)
 #define SNDRV_SST_STREAM_GET_PARAMS _IOWR('L', 0x01,   struct snd_sst_get_stream_params *)
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
-- 
1.7.4.1

